job-descriptions/software-engineer-backend.md

```diff
@@ -0,0 +1,61 @@
+![logo](https://sourcegraph.com/.assets/img/sourcegraph-light-head-logo.svg)
+
+# Software Engineer - Backend
+
+### About us
+
+We're building the new standard developer platform. Top tech companies have invested \$100Ms to build internal developer platforms for code search, code review, alerts, and automation. Sourcegraph provides this standard developer platform to every company, helping startups and large enterprises ship better software faster.
+
+[Our mission](https://sourcegraph.com/plan) is to dramatically increase the number of people who can understand and write code. By making code more accessible, we will democratize software development and accelerate innovations that bring the future sooner in transportation, health care, energy, AI, communication, space travel, etc.
+
+We value openness and transparency. [Our code](https://github.com/sourcegraph/sourcegraph), [our roadmap](https://about.sourcegraph.com/direction), and [our company processes](https://about.sourcegraph.com/company/open_source_open_company) are public. This transparency helps us rapidly gather feedback from our customers so we can iterate, learn, and deliver the best product.
+
+To create a product that serves the needs of all developers, we are building a diverse remote-first team that is distributed across the world. Sourcegraph is an equal opportunity workplace; we welcome people from all backgrounds and communities.
+
+If you are passionate about making the world better through software, come join us!
+
+### About the role
+
+As a backend engineer at Sourcegraph, you will focus on designing and developing fast and reliable services that power our product (e.g., code search, code intelligence, automation). Read our [project roadmap](https://docs.google.com/document/d/1cBsE9801DcBF9chZyMnxRdolqM_1c2pPyGQz15QAvYI/edit#heading=h.g2wq4qci7wj0) to see the kinds of projects that we are working on today and have planned for the future.
+
+We are looking for candidates with the following skills and a strong desire to learn:
+
+- Excellent knowledge of Go (preferred) or TypeScript.
```

Comment:
If this job description is also used to attract code intel candidates, I think the "excellent knowledge of Go (required)" might be a bit too much, since they're happy to accept generalists.

Reply:
Structurally I think it is important and valuable to separate requirements from nice-to-have.

@tsenart do you want to add Go as a requirement because it is a requirement for the core services team or because you think even members of the code intel team should be required to know Go?

@efritz @chrismwendt do you think code intel candidates should be required to know Go? Alternatively do you think they should be require to know Go OR TypeScript? More generally, how would you define the requirements for code intel teammates?

If Go is a requirement for core services but not for code intel, then we will probably need separate job descriptions.

Reply:
I am going to work on splitting these job descriptions.

Reply:
I created a separate job description for code intel team because they didn't feel like Go was a strict requirement for them. If there are any further comments about this, please start a new thread on the relevant section of the new job description.